### ISSUE TITLE
Bug in api.query.system.events

### ISSUE THREAD

**Issue by mariopino** [polkadot-js/api#1873]
Strange! Can be reproduced in https://polkadot.js.org/apps/#/js:

```
const block = 3876;
const blockHash = await api.rpc.chain.getBlockHash(block);
const blockEvents = await api.query.system.events.at(blockHash);
```
Result:

```
Error: getStorage(key: StorageKey, at?: BlockHash): StorageData:: createType(Vec<EventRecord>):: Unable to create Enum via index 5, in ApplyExtrinsic, Finalization
    at e.selector (https://polkadot.js.org/apps/main.5fcc7456.js:65:323920)
    at e.error (https://polkadot.js.org/apps/vendor.af33e3c6.js:1:57594)
    at e._next (https://polkadot.js.org/apps/vendor.af33e3c6.js:8:20912)
    at e.next (https://polkadot.js.org/apps/vendor.af33e3c6.js:1:971)
    at e.notifyNext (https://polkadot.js.org/apps/vendor.af33e3c6.js:1:95724)
    at e._next (https://polkadot.js.org/apps/vendor.af33e3c6.js:8:22338)
    at e.next (https://polkadot.js.org/apps/vendor.af33e3c6.js:1:971)
    at e.notifyNext (https://polkadot.js.org/apps/vendor.af33e3c6.js:8:88172)
    at e._next (https://polkadot.js.org/apps/vendor.af33e3c6.js:8:22338)
    at e.next (https://polkadot.js.org/apps/vendor.af33e3c6.js:1:971)
```

**Comment by mariopino** [comment]
That block also breaks Polkadot JS UI:

https://polkadot.js.org/apps/#/explorer/query/3876

Event with index 5 doesn't exist in PolkaScan:

https://polkascan.io/pre/kusama/block/3876

**Comment by jacogr** [comment]
You cannot query historic events this way -

- the API only decorated the latest metadat
- the types are outdated at that point

See https://github.com/polkadot-js/api/blob/master/packages/types/src/known/overrides.ts#L50 & https://github.com/polkadot-js/api/issues/845

TL;DR The API is a real-time tool, atm for historical queries, you are on your own. (The types and metadata for historical blocks need to be injected). There may be an "archive mode" at some point, however at this point querying both the state and the metadata at the previous block (metadat takes effect at the end of the block), it out of scope.
